src/main/kotlin/com/expedia/graphql/schema/SchemaGenerator.kt

```diff
@@ -291,12 +299,28 @@ internal class SchemaGenerator(
         builder.typeResolver { env: TypeResolutionEnvironment -> env.schema.getObjectType(env.getObject<Any>().javaClass.simpleName) }
         val interfaceType = builder.build()
 
-        val reflections = Reflections(config.supportedPackages)
         val implementations = reflections.getSubTypesOf(Class.forName(kClass.javaObjectType.name))
         implementations.forEach {
             additionTypes.add(objectType(it.kotlin, interfaceType))
         }
 
         return interfaceType
     }
+
+    private fun unionType(kClass: KClass<*>): GraphQLType {
+        val builder = GraphQLUnionType.newUnionType()
+
+        builder.name(kClass.simpleName)
+        builder.description(kClass.graphQLDescription())
+        builder.typeResolver { env: TypeResolutionEnvironment -> env.schema.getObjectType(env.getObject<Any>().javaClass.simpleName) }
+
+        val implementations = reflections.getSubTypesOf(Class.forName(kClass.javaObjectType.name))
+        implementations
+            .forEach {
+                builder.possibleType(objectType(it.kotlin) as GraphQLObjectType)
```

Comment:
dont you have to add it to the `additionTypes`?

Reply:
you don't have to as `objectType` adds the type to schema directly